_posts/2023-02-22-user-defined-operators.md

```diff
@@ -0,0 +1,598 @@
+date: '2023-02-22'
+author: Kamil Śliwak, Matheus Aguiar
+category: Explainers
+---
+
+The prominent new feature in [Solidity 0.8.19](https://github.com/ethereum/solidity/releases/tag/v0.8.19)
+is the support for defining *operators* on [user-defined value types](https://docs.soliditylang.org/en/latest/types.html#user-defined-value-types) (UDVTs).
+
+If you have not been keeping up with recent features, UDVTs are a new class of types introduced in Solidity 0.8.8.
+They provide an abstraction over an elementary value type that results in a completely new type.
+This is similar to creating an alias, but the new type is distinct from the underlying value type
+and all other UDVTs derived from that underlying type.
+The ability to use operators is meant to bring the UDVTs closer to being as natural to use as the
+built-in types and promote the use of type-safe interfaces.
+Below, we are going to take a closer look into the details of this new mechanism.
+
+### A Quick Refresher
+Before we get into the details of the new feature and how it affects the language, it may be a good
+idea to first go over the two existing features that it builds upon: built-in operators and `using for`.
+
+#### Built-in operators
+Solidity, like the vast majority of programming languages,
+[offers a set of operators](https://docs.soliditylang.org/en/latest/types.html#operators) as syntactic sugar
+over common arithmetical and logical functions.
+The difference between operators and functions is purely syntactical, and the decision
+to choose one over the other is to a large extent a matter of readability.
+The use of operators makes expressions more concise, which is helpful as long as they are well
+understood by the reader.
+What is included in that set varies widely from language to language.
+
+Here is the full list for Solidity:
+
+| Category             | Operators                                                                |
+|----------------------|--------------------------------------------------------------------------|
+| Bitwise              | `&`, `\|`, `^`, `<<`, `>>`, `~`                                          |
+| Arithmetic           | `+`, `-`, `*`, `/`, `%`, `**`                                            |
+| Comparison           | `==`, `!=`, `<`, `<=`, `>`, `>=`                                         |
+| Boolean              | `!`, `&&`, `\|\|`                                                        |
+| Increment/decrement  | `++`, `--`                                                               |
+| Simple assignment    | `=`                                                                      |
+| Composite assignment | `+=`, `-=`, `^=`, `*=`, `/=`, `%=`, `&=`, `\|=`, `^=`, `<<=`, `>>=`      |
+| Miscellaneous        | `delete`, `new`, `.`, `()`, `[]`, `?:`                                   |
+
+It is important to note that in many languages, an assignment is a statement that is handled specially.
```

Comment:
```suggestion
It is important to note that in many languages, an assignment is a statement that is handled uniquely.
```

"Specially" sounds odd to me - maybe rather ``uniquely`` or ``distinctively``?

Reply:
Odd, it sounded fine to me. I think I see it quite often. It's `uniquely` and `distinctively` that seem off to me in this context :)

But `in a special way` is good. I will change to that.